### ISSUE TITLE
Alerting - failed to enable alerting in new cluster after upgrading Rancher 

### ISSUE THREAD

**Issue by jiaqiluo** [rancher/rancher#21399]
**What kind of request is this (question/bug/enhancement/feature request):**
Bug, found during validating this issue: https://github.com/rancher/rancher/issues/21364

**Steps to reproduce (least amount of steps as possible):**
- run Rancher: v2.2.4 container
- as admin, enable Github auth
- log in as Github user user1
- add the following three clusters: DO single-node, DO 3-node, and GKE
- adds cluster notifier and cluster alerting for all three clusters
- upgrade Rancher to v2.2.5-rc9
- adding a new single-node cluster and a new 3-node cluster in the same setup and enabling the cluster alerting

**Result:**
- FAILED to enable the cluster alerting in newly-added clusters
- Rancher shows the following errors

```
Failed to ensure monitoring project name: failed to move Namespace cattle-prometheus into a Project: Operation cannot be fulfilled on namespaces "cattle-prometheus": the object has been modified; please apply your changes to the latest version and try again
```
- Rancher's log shows
 
```
2019/07/08 21:31:58 [ERROR] ClusterAlertGroupController c-28fpf/cag-8cwmg [cluster-alert-group-deployer] failed with : failed to get Alertmanager Deployment information, statefulsets.apps "alertmanager-cluster-alerting" not found, [cluster-alert-group-controller] failed with : Failed to get service for alertmanager, service "cattle-prometheus/alertmanager-operated" not found
2019/07/08 21:32:00 [ERROR] ClusterAlertRuleController c-28fpf/car-8fw8v [cluster-alert-rule-deployer] failed with : failed to get Alertmanager Deployment information, statefulsets.apps "alertmanager-cluster-alerting" not found, [cluster-alert-rule-controller] failed with : Failed to get service for alertmanager, service "cattle-prometheus/alertmanager-operated" not found
2019/07/08 21:32:04 [ERROR] ClusterAlertGroupController c-28fpf/cag-8cwmg [cluster-alert-group-deployer] failed with : failed to get Alertmanager Deployment information, statefulsets.apps "alertmanager-cluster-alerting" not found, [cluster-alert-group-controller] failed with : Failed to get service for alertmanager, service "cattle-prometheus/alertmanager-operated" not found
```

<img width="1408" alt="screenshot" src="https://user-images.githubusercontent.com/6218999/60845866-7ca50f00-a192-11e9-8dd4-09d8c849480a.png">

**Comment by jiaqiluo** [comment]
I met the same error in a fresh-installed Rancher v2.2.3 setup

<img width="1526" alt="Screen Shot 2019-07-09 at 4 25 05 PM" src="https://user-images.githubusercontent.com/6218999/60930241-9a917300-a269-11e9-9be5-9dd1b7c1bddf.png">

And the workaround: 
- step 1:  delete the alter group, which triggers the deletion of the apps `cluster-alerting` and `monitoring-operator` if they exist
- step 2: add a new alter group, new alter rules, set the notifier as the receiver, which triggers the re-deployment of those apps

It could fail with the same errors. If so, repeat those 2 steps until it succeeds.

**Comment by jiaqiluo** [comment]
I met the same error in a newly-added DO single-node cluster in the setup that upgrads from v2.2.3 to v2.2.5-rc12

**Comment by jiaqiluo** [comment]
failed to validate the bug fix in the setup v2.2.8 upgraded to v2.3.0-rc6

The cluster alerting is enabled successfully, the apps `cluster-alerting` and `monitoring-operator` are deployed, but there are no alerting being triggered and messages being sent out when the condition happens.

**Comment by jiaqiluo** [comment]
It turns out I set the condition of the event wrong, so the alert is never trigged. 

After fixing that, the alerts are sent out successfully. 

The bug fix is validated now.
